### ISSUE TITLE
Implement combined and sequential skills based on design documentation

### ISSUE THREAD

**Issue by anchristensen** [uchicago-cs/chiventure#1278]
Similarly to the design warmup, in issue #1230 we added a header file for complex skills and added supporting structs as well as functions to be implemented later. In this issue, we will actually implement the functions we prepared by 1) filling out the functions in a .c file and 2) editing some of the functions in the skill module to support the changes to the skill struct we made. This will finish the implementation of combined and sequential skills.

**Comment by anchristensen** [comment]
Brief update: All of the function headers are written and the majority of the body of the functions in the .c file filled out. We still need to write out testing of the functions and finish two function bodies (complex_skill_level_up and complex_skill_xp_up) to make sure their functionality makes sense. Then a pull request will be submitted.

**Comment by anchristensen** [comment]
Closing statement: a pull request was just created for this issue https://github.com/uchicago-cs/chiventure/pull/1382. We filled in the body of the functions we wrote as part of the design process, including a new, init, free, xp_up, level_up, and execute function for complex skills. Tests have also been added in a new complex_skill_test file for each of these functions. @nbilaver wrote most of the new, init, free, and execute functions and tests for them, and I wrote the xp_up and level_up functions, the sequential skill execute function, and tests for those functions. We both worked on debugging.
